core/src/main/java/org/frankframework/jta/narayana/NarayanaDataSourceFactory.java

```diff
@@ -15,104 +15,59 @@
 */
 package org.frankframework.jta.narayana;
 
-import java.sql.Connection;
-import java.time.Duration;
+import static java.util.Objects.requireNonNull;
 
-import javax.sql.CommonDataSource;
 import javax.sql.DataSource;
 import javax.sql.XADataSource;
 
-import org.apache.commons.dbcp2.ConnectionFactory;
-import org.apache.commons.dbcp2.DataSourceConnectionFactory;
 import org.apache.commons.dbcp2.PoolableConnection;
 import org.apache.commons.dbcp2.PoolableConnectionFactory;
-import org.apache.commons.dbcp2.PoolingDataSource;
 import org.apache.commons.dbcp2.managed.DataSourceXAConnectionFactory;
 import org.apache.commons.dbcp2.managed.PoolableManagedConnectionFactory;
 import org.apache.commons.dbcp2.managed.XAConnectionFactory;
-import org.apache.commons.pool2.ObjectPool;
 import org.apache.commons.pool2.impl.GenericObjectPool;
 import org.frankframework.jdbc.datasource.OpenManagedDataSource;
-import org.frankframework.jndi.JndiDataSourceFactory;
+import org.frankframework.jndi.AbstractXADataSourceFactory;
 import org.frankframework.util.AppConstants;
 
 import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
 
-import lombok.Getter;
 import lombok.Setter;
 import lombok.extern.log4j.Log4j2;
 
 @Log4j2
-public class NarayanaDataSourceFactory extends JndiDataSourceFactory {
+public class NarayanaDataSourceFactory extends AbstractXADataSourceFactory {
 
-	private @Getter @Setter int minPoolSize = 0;
-	private @Getter @Setter int maxPoolSize = 20;
-	private @Getter @Setter int maxLifeTimeSeconds = 0;
+	private @Setter NarayanaJtaTransactionManager transactionManager;
 
 	public NarayanaDataSourceFactory() {
+		// For backwards compatibility, apply these configuration constants if they're found.
 		AppConstants appConstants = AppConstants.getInstance();
 		minPoolSize = appConstants.getInt("transactionmanager.narayana.jdbc.connection.minPoolSize", minPoolSize);
 		maxPoolSize = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxPoolSize", maxPoolSize);
-		maxLifeTimeSeconds = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxLifeTime", maxLifeTimeSeconds);
+		maxIdle = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxIdle", maxIdle);
+		maxLifeTime = appConstants.getInt("transactionmanager.narayana.jdbc.connection.maxLifeTime", maxLifeTime);
+		testQuery = appConstants.getString("transactionmanager.narayana.jdbc.connection.testQuery", testQuery);
```

Comment:
All these extra/new features we set are valuable improvements, to my humble opinion, even though perhaps Ali's main issue was solved differently.